query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -341,4 +347,85 @@ private QueryRequest rebuildRequestOmittingDefaults(
         .addAllOrderBy(orderBys)
         .build();
   }
+
+  // We need the CONTAINS_KEY filter in all filters and order bys dealing with complex
```

Comment:
nit: Can we add some more comments?
It looks like we have preferences, if the contain_key is added by orderBy expression, we are not adding the same while checking on filtering expression. That looks fine. But, can we add this information in comment?

Reply:
resolved